synapse/handlers/search.py

```diff
@@ -421,10 +422,10 @@ async def search(
 
         for context in contexts.values():
             context["events_before"] = self._event_serializer.serialize_events(
-                context["events_before"], time_now
+                context["events_before"], time_now  # type: ignore[arg-type]
             )
             context["events_after"] = self._event_serializer.serialize_events(
-                context["events_after"], time_now
+                context["events_after"], time_now  # type: ignore[arg-type]
             )
```

Comment:
Yes, this still mutates the dictionary, but it seems rather messy to *not* do this...